### PR TITLE
[ADN] Implement login

### LINKED ISSUE #27841: [ADN] Can't download premium video after extractor fix
## Checklist

- [x] I'm asking a question
- [x] I've looked through the README and FAQ for similar questions
- [x] I've searched the bugtracker for similar questions including closed ones

## Question

First of all I would like to thank the person who fixed the ADN extractor.

Everything works perfectly to download the free videos in 480p. But to download videos requiring a premium account or to download better quality I can't do it.

Before I used to use the command :

`youtube-dl.exe --cookies ./cookies.txt https://animedigitalnetwork.fr/video/my-hero-academia-saison-4/10271-episode-2-overhaul`

And it worked perfectly but after the extractor fix it doesn't work anymore I tried to re-download the cookies as indicated in the doc but it doesn't change anything I also tried to use `--username` and `--password` but it doesn't do anything either and it shows me in the both case:

`ERROR: This video is only available for paying users`

I think the problem comes from the cookies that don't have the same shape as before, there were 4 lines in my file but now there is only one.

Could someone help me?

Thank you.

### PR DESCRIPTION
**Pull request by adrianheine**
### Before submitting a *pull request* make sure you have:
- [x] [Searched](https://github.com/ytdl-org/youtube-dl/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Read [adding new extractor tutorial](https://github.com/ytdl-org/youtube-dl#adding-support-for-a-new-site)
- [x] Read [youtube-dl coding conventions](https://github.com/ytdl-org/youtube-dl#youtube-dl-coding-conventions) and adjusted the code to meet them
- [x] Covered the code with tests (note that PRs without tests will be REJECTED)
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8)

### In order to be accepted and merged into youtube-dl each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [ ] Bug fix
- [x] Improvement
- [ ] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information

Closes #27841.

### CODE REVIEW

File: youtube_dl/extractor/adn.py [hunk 1]
```diff
@@ -51,6 +52,7 @@ class ADNIE(InfoExtractor):
         }
     }
 
+    _NETRC_MACHINE = 'animedigitalnetwork.fr'
```
Comment (remitamine) [inline youtube_dl/extractor/adn.py:55]:
```suggestion
    _NETRC_MACHINE = 'animedigitalnetwork'
```

File: youtube_dl/extractor/adn.py [hunk 2]
```diff
@@ -132,16 +134,30 @@ def _get_subtitles(self, sub_url, video_id):
...
+                'Logging in',
+                data=urlencode_postdata({
+                    'password': password,
+                    'rememberMe': False,
+                    'source': "Web",
+                    'username': username,
+                })).get('accessToken')
+            if accessToken:
+                headers = {'authorization': 'Bearer ' + accessToken}
+
```
Comment (remitamine) [inline youtube_dl/extractor/adn.py:152]: should be done once when the extractor is initialized.

**Review by remitamine — CHANGES_REQUESTED** [review]

File: youtube_dl/extractor/adn.py [hunk 3]
```diff
@@ -129,19 +132,32 @@ def _get_subtitles(self, sub_url, video_id):
             }])
         return subtitles
 
+    def _real_initialize(self):
+        username, password = self._get_login_info()
+        if username:
+            accessToken = self._download_json(
+                self._API_BASE_URL + "authentication/login", "animedigitalnetwork.fr",
+                'Logging in',
+                data=urlencode_postdata({
```
Comment (remitamine) [inline youtube_dl/extractor/adn.py:141]:
```suggestion
                self._API_BASE_URL + 'authentication/login', None,
                'Logging in', data=urlencode_postdata({
```

**Review by remitamine — CHANGES_REQUESTED** [review]
extraction should not stop if login process fails.

**Comment by MrTechyyy** [comment]
Thanks for the fix, really appreciate it!

**Comment by Azzuray** [comment]
Download now works fine with --username --password 💹  but subtittle decrypt doesnt work ❌

**Comment by adrianheine** [comment]
> extraction should not stop if login process fails.

You sure? I think if I provide login credentials but they are wrong I wouldn't expect extraction to silently continue, given that not being logged in might mean lower quality.

**Comment by remitamine** [comment]
> > extraction should not stop if login process fails.
> 
> You sure? I think if I provide login credentials but they are wrong I wouldn't expect extraction to silently continue, given that not being logged in might mean lower quality.

not silently, there would be a warning.

**Comment by adrianheine** [comment]
Hmm … I think I'd rather have people remove their login credentials if there's a problem but I can also agree with a warning.

File: youtube_dl/extractor/adn.py [hunk 4]
```diff
@@ -129,19 +132,31 @@ def _get_subtitles(self, sub_url, video_id):
...
+        username, password = self._get_login_info()
+        if username:
+            accessToken = (self._download_json(
+                self._API_BASE_URL + "authentication/login", None,
+                'Logging in', data=urlencode_postdata({
+                    'password': password,
+                    'rememberMe': False,
+                    'source': "Web",
+                    'username': username,
+                }), fatal=False, errnote="Unable to log in") or {}).get('accessToken')
```
Comment (remitamine) [inline youtube_dl/extractor/adn.py:145]: - use single quotes consistently.
- i think it would be better to keep errnote closer to note.

**Review by remitamine — CHANGES_REQUESTED** [review]
an account is needed to check the code locally.

File: youtube_dl/extractor/adn.py [hunk 5]
```diff
@@ -129,19 +132,32 @@ def _get_subtitles(self, sub_url, video_id):
             }])
         return subtitles
 
+    def _real_initialize(self):
+        username, password = self._get_login_info()
+        if username:
+            accessToken = (self._download_json(
```
Comment (remitamine) [inline youtube_dl/extractor/adn.py:138]: use the common Python variable names convension.

**Pull request merged**